OpenRA.Mods.Common/Traits/Air/Aircraft.cs

```diff
@@ -868,9 +857,53 @@ public Activity MoveTo(CPos cell, Actor ignoreActor, Color? targetLineColor = nu
 				initialTargetPosition, targetLineColor);
 		}
 
-		public Activity MoveIntoWorld(Actor self, CPos cell, SubCell subCell = SubCell.Any)
+		public Activity MoveIntoWorld(Actor self, int delay = 0)
+		{
+			return new MoveIntoWorldActivity(self, delay);
+		}
+
+		class MoveIntoWorldActivity : Activity
 		{
-			return new Fly(self, Target.FromCell(self.World, cell, subCell));
+			readonly Actor self;
+			readonly Aircraft aircraft;
+			readonly int delay;
+
+			public MoveIntoWorldActivity(Actor self, int delay = 0)
+			{
+				this.self = self;
+				aircraft = self.Trait<Aircraft>();
+				IsInterruptible = false;
```

Comment:
If the actor can't make it to the rallypoint (e.g. is blocked by repulsion) this causes the helicopter to be stuck uncontrollable. Simple repro case: enable cheats and build five helicopters at once.

Reply:
Fixed as discussed on IRC